src/api/ledger/parse/pathfind.js

```diff
@@ -8,17 +8,25 @@ function parsePaths(paths) {
     _.omit(step, ['type', 'type_hex'])));
 }
 
+function ensureIOUCounterparty(amount, sourceAmount) {
+  const defaults = sourceAmount.counterparty ?
+    {counterparty: sourceAmount.counterparty} : {};
+  return amount.currency === 'XRP' || amount.counterparty ? amount :
+    _.assign(defaults, amount);
+}
+
 function parsePathfind(sourceAddress: string,
     destinationAmount: Object, pathfindResult: Object): Object {
   return pathfindResult.alternatives.map(function(alternative) {
+    const sourceAmount = parseAmount(alternative.source_amount);
     return {
       source: {
         address: sourceAddress,
-        amount: parseAmount(alternative.source_amount)
+        amount: sourceAmount
       },
       destination: {
         address: pathfindResult.destination_account,
-        amount: destinationAmount
+        amount: ensureIOUCounterparty(destinationAmount, sourceAmount)
```

Comment:
The `destination.amount` is what's used for the `tx_json.Amount` right?

This doesn't look quite right as elsewhere:
```js
if (typeof params.dst_amount === 'object' && !params.dst_amount.issuer) {
    // Convert blank issuer to sender's address
    // (Ripple convention for 'any issuer')
    // https://ripple.com/build/transactions/
    //     #special-issuer-values-for-sendmax-and-amount
    // https://ripple.com/build/ripple-rest/#counterparties-in-payments
    params.dst_amount.issuer = params.dst_account;
  }
```

It uses the destination account as default counterparty, rather than the source amounts'